config/initializers/devise.rb

```diff
@@ -113,6 +113,319 @@
+
+  # Configure which authentication keys should have whitespace stripped.
+  # These keys will have whitespace before and after removed upon creating or
+  # modifying a user and when used to authenticate or find a user. Default is :email.
+  config.strip_whitespace_keys = [:email]
+
+  # Tell if authentication through request.params is enabled. True by default.
+  # It can be set to an array that will enable params authentication only for the
+  # given strategies, for example, `config.params_authenticatable = [:database]` will
+  # enable it only for database (email + password) authentication.
+  # config.params_authenticatable = true
+
+  # Tell if authentication through HTTP Auth is enabled. False by default.
+  # It can be set to an array that will enable http authentication only for the
+  # given strategies, for example, `config.http_authenticatable = [:database]` will
+  # enable it only for database authentication.
+  # For API-only applications to support authentication "out-of-the-box", you will likely want to
+  # enable this with :database unless you are using a custom strategy.
+  # The supported strategies are:
+  # :database      = Support basic authentication with authentication key + password
+  # config.http_authenticatable = false
+
+  # If 401 status code should be returned for AJAX requests. True by default.
+  # config.http_authenticatable_on_xhr = true
+
+  # The realm used in Http Basic Authentication. 'Application' by default.
+  # config.http_authentication_realm = 'Application'
+
+  # It will change confirmation, password recovery and other workflows
+  # to behave the same regardless if the e-mail provided was right or wrong.
+  # Does not affect registerable.
+  # config.paranoid = true
+
+  # By default Devise will store the user in session. You can skip storage for
+  # particular strategies by setting this option.
+  # Notice that if you are skipping storage for all authentication paths, you
+  # may want to disable generating routes to Devise's sessions controller by
+  # passing skip: :sessions to `devise_for` in your config/routes.rb
+  config.skip_session_storage = [:http_auth]
+
+  # By default, Devise cleans up the CSRF token on authentication to
+  # avoid CSRF token fixation attacks. This means that, when using AJAX
+  # requests for sign in and sign up, you need to get a new CSRF token
+  # from the server. You can disable this option at your own risk.
+  # config.clean_up_csrf_token_on_authentication = true
+
+  # When false, Devise will not attempt to reload routes on eager load.
+  # This can reduce the time taken to boot the app but if your application
+  # requires the Devise mappings to be loaded during boot time the application
+  # won't boot properly.
+  # config.reload_routes = true
+
+  config.navigational_formats = ['*/*', :html, :turbo_stream]
+
   config.navigational_formats = ['*/*', :html, :turbo_stream]
 
   # ==> Configuration for :database_authenticatable
```

Comment:
There are two configs for the same thing :) The _Configuration for :database_authenticatable_ is written twice in this file. I was wondering about the diff here as in how on earth is this new code, but now I know :D

Reply:
I am speechless :O    Check :)

Reply:
I'm not sure if it's checked - it's still in the diff 😅

Reply:
No it's not :)

Reply:
I'm afraid it's even worse - whole Devise config is duplicated! Attaching screenshots :) 
<img width="1352" alt="image" src="https://user-images.githubusercontent.com/5008759/214955887-f4be7f82-aa73-44ec-9183-1c59a4cfef59.png">

Then

<img width="1359" alt="image" src="https://user-images.githubusercontent.com/5008759/214956009-4153cf9f-3e05-4a76-a6a9-c955f1ede4fc.png">


and.. there's another one, in same file!

<img width="1360" alt="image" src="https://user-images.githubusercontent.com/5008759/214956082-12e38cca-79b7-4ac0-87a7-eb1f759a9bfd.png">

I'm actually amazed that this works, but when I think about it makes sense - second `Devise.setup` call must be reseting the first one! 🤯

Reply:
I've messed up this one. Sorry. Check :)

Reply:
No worries, happens to the best of us 👍 Good job fixing!